### PR TITLE
skin preview screenshots in Interface preferences

### PR DESCRIPTION
**Pull request by Be-ing**
#1377 implemented standard behavior for the skin selection combobox so it only changes the skin when the user presses the Apply button. To provide users a rough impression of what skins look like before they switch to a different skin, show a preview screenshot when a skin is selected with the combobox.

### CODE REVIEW

File: src/preferences/dialog/dlgprefinterfacedlg.ui
Comment (esbrandt) [inline src/preferences/dialog/dlgprefinterfacedlg.ui]: String changed by accident?

File: src/preferences/dialog/dlgprefinterfacedlg.ui
Comment (esbrandt) [inline src/preferences/dialog/dlgprefinterfacedlg.ui]: Dummy text can be excluded from the translation template.
``<string notr="true">skin preview screenshot goes here</string>``

File: src/preferences/dialog/dlgprefinterfacedlg.ui
Comment (esbrandt) [inline src/preferences/dialog/dlgprefinterfacedlg.ui]: String changed by accident?

File: src/skin/skinloader.cpp
Comment (esbrandt) [inline src/skin/skinloader.cpp]: What about a more descriptive file name, e.g. ``preferences_skin_preview.png`` or ``preferences_skin_screenshoot.png``?

**Review by esbrandt — CHANGES_REQUESTED** [review]
Thanks for working on this long pending feature.

I feel the screenshoots should scale down at least to the same height. When flipping trough the list of available skins, the options below the screenshoots jump around.

The Deere screenshoot has a good size, it could be even a bit bigger IMO (~~400px~~ Edit: 200px). Hard to make out some details with the smaller screenshoots, probably even harden when using HDPi displays.

Also, should we provide the screenshoots in the vanilla skin state? E.g. in the Deere screen, stacked waveforms are visible, but are not enabled by default.

File: src/preferences/dialog/dlgprefinterfacedlg.ui
Comment (Be-ing) [reply src/preferences/dialog/dlgprefinterfacedlg.ui]: QtDesigner did this automatically. I can rebase and remove the noise from the diff.

**Comment by Be-ing** [comment]
> I feel the screenshoots should scale down at least to the same height. When flipping trough the list of available skins, the options below the screenshoots jump around.

Yes, I will redo the screenshots...

> Also, should we provide the screenshoots in the vanilla skin state?

Yes, good point. I will redo the screenshots with a fresh settings directory.

> E.g. in the Deere screen, stacked waveforms are visible, but are not enabled by default.

Parallel waveforms are the default in Deere.

> The Deere screenshoot has a good size, it could be even a bit bigger IMO (~~400px~~ Edit: 200px). Hard to make out some details with the smaller screenshoots, probably even harden when using HDPi displays.

IMO details are unimportant for this. I do not want to make the screenshots so big that they clutter the preferences window. I'll try a height of 300 px.

**Comment by daschuer** [comment]
Shouldn't this go to master?

**Comment by Be-ing** [comment]
It doesn't matter too much for me, but I thought you were [bothered by the current behavior](https://bugs.launchpad.net/mixxx/+bug/1745684). IMO targeting this to master would be overly pedantic about feature freeze. The code changes are really trivial and clearly won't break anything else. OTOH reverting to the 2.0 behavior of loading skins as soon as they are selecting (before pressing Apply) would be more difficult than merging this.

**Comment by daschuer** [comment]
OK, than let's keep this for 2.1

**Comment by nopeppermint** [comment]
nice, maybe you could add also a preview of the color schemes in Shade

**Comment by daschuer** [comment]
> I feel the screenshoots should scale down at least to the same height. When flipping trough the list of available skins, the options below the screenshoots jump around.

Yes, that is annoying. IMHO the screenshots should be all at the same width as well.
A smaller size like the current Shade screenshot would be nice. It would work to me, to how only the top right part, to find a compromise between screen shot size and down scaling. 

We need also a default screenshot for skins without a screenshot.

**Comment by Be-ing** [comment]
Yes. I'll update the screenshots when all pending skin PRs are merged.

**Comment by daschuer** [comment]
So this PR should only include the default screenshot, right? 
Can you rebase this PR to not include the outdated screenshots into the git history?

**Comment by Be-ing** [comment]
Yes, this PR should only show the skins in the default configurations. I'll rebase it when the time comes.

**Comment by daschuer** [comment]
I mean, we can merge this now, if you remove the real screenshots and add a placeholder for any skin without a screenshot.

**Comment by daschuer** [comment]
@Be-ing is this still a 2.1 candidate? 
Can you add the current screen shots by rebasing this, to keep this PR small sized? 
Can you do the screenshots with 100 % scaling to not show scaling issues if any?

**Comment by daschuer** [comment]
They should be taken with design size.
As of Shade 1024*600 
Resize to 300 is already quite high. How about 250

**Comment by Be-ing** [comment]
We can't use Shade's minimum size for the screenshots because not all skins support that. We could use the lowest common denominator size of all skins.

**Comment by daschuer** [comment]
Yes that's right. I mean to use the optimum screen size for each skin. I would say it is full HD for LateNight for example. For Tango, we can decide if we want to show the netbook mode or the 4 deck full scale mode.

**Comment by ronso0** [comment]
> For Tango, we can decide if we want to show the netbook mode or the 4 deck full scale mode.

I propose 1280px or 1366px with default configuration: 50px waveforms, 2 decks, 2 FX units, ...

**Comment by daschuer** [comment]
src/preferences/dialog/dlgprefinterfacedlg.ui
uic: Error in line 241, column 42 : Unexpected attribute notr
File 'src/preferences/dialog/dlgprefinterfacedlg.ui' is not valid

**Comment by daschuer** [comment]
Ping! An other day delay.
We need a rc now. Will you have time to finish this today?

**Comment by daschuer** [comment]
Placeholder:

[placeholder](https://user-images.githubusercontent.com/1777442/38362016-a737c262-38cf-11e8-97f5-ad7f61902b58.png)

**Comment by Be-ing** [comment]
Don't merge quite yet, still a few little quirks to fix in the code.

**Comment by daschuer** [comment]
Please use the optimum size for the screenshots, not a unique size.
This shows how the skin is intended to use.
Which is the mayor use case for this preview feature.
This lines also quite well up, if we have one day a tiny skin, or a full HD only skin.

**Comment by Be-ing** [comment]
The screenshots show the default skin configurations.

**Comment by Be-ing** [comment]
The "(64 sampler)" skins are not showing the screenshot for their parent skin. Do you have any hints how to make that work?

**Comment by Be-ing** [comment]
Nevermind, I'll take screenshots showing 64 samplers for those.

**Comment by Be-ing** [comment]
Okay, done.

**Pull request merged**
